test/middleware/CompanyAuthMiddleware.test.ts

```diff
@@ -0,0 +1,155 @@
+import 'reflect-metadata';
+
+import { Arg, Substitute, SubstituteOf } from '@fluffy-spoon/substitute';
+import { Session, SessionStore } from 'ch-node-session-handler';
+import { SessionKey } from 'ch-node-session-handler/lib/session/keys/SessionKey';
+import { ISignInInfo } from 'ch-node-session-handler/lib/session/model/SessionInterfaces';
+import { expect } from 'chai';
+import { NextFunction, Request, Response } from 'express';
+
+import 'app/controllers/index';
+import { CompanyAuthMiddleware } from 'app/middleware/CompanyAuthMiddleware';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { CompanyAuthConfig } from 'app/models/CompanyAuthConfig';
+import { JwtEncryptionService } from 'app/modules/jwt-encryption-service/JwtEncryptionService';
+import { SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+
+import { createSubstituteOf } from 'test/SubstituteFactory';
+
+describe('Company Authentication Middleware', () => {
+
+    const appeal: Appeal = {
+        penaltyIdentifier: {
+            companyNumber: 'SC123123',
+            userInputPenaltyReference: 'A1231234',
+            penaltyReference: 'A1231234'
+        },
+        reasons: {
+            other: {
+                title: '',
+                description: ''
+            }
+        }
+    };
+
+    const companyAuthConfig: CompanyAuthConfig = {
+        oathScopePrefix: 'MOCK',
+        accountUrl: 'MOCK',
+        accountRequestKey: 'MOCK',
+        accountClientId: 'MOCK',
+        chsUrl: 'MOCK',
+    };
+
+    const sessionStoreForAuthConfig = {
+        sessionCookieName: '__SID',
+        sessionCookieDomain: 'rebel1.aws.chdev.org',
+        sessionCookieSecureFlag: 'true',
+        sessionTimeToLiveInSeconds: 3600
+    };
+
+    const featureFlag: string = '1';
+
+    it('should call next with an error when session is undefined', async () => {
+
+        const encryptionService = createEncryptionService('resolves');
+
+        const sessionStore = Substitute.for<SessionStore>();
+        const companyAuthMiddleware = new CompanyAuthMiddleware(
+            sessionStore,
+            encryptionService,
+            companyAuthConfig,
+            sessionStoreForAuthConfig,
+            featureFlag
+        );
+
+        const nextFunction = createSubstituteOf<NextFunction>();
+        const response = createSubstituteOf<Response>();
+        const request = { session: undefined } as Request;
+
+        try{
+            await companyAuthMiddleware.handler(request, response, nextFunction);
+        } catch (err) {
+            expect(err.message).to.equal(SESSION_NOT_FOUND_ERROR.message);
```

Comment:
How does it work? Test says that next should be called while exception is being caught. I thought you will do verification of what arg next function was invoked with.